### PR TITLE
Fix various issues regarding variants

### PR DESCRIPTION
**Pull request by remcohaszing**
Mostly some code has been shuffled and typed to make both ESLint and TypeScript happy.

### CODE REVIEW

**Comment by mhsdesign** [comment]
the main branch is ahead with 3 commits (1 is unrelated ..)

in those two commits:
https://github.com/remcohaszing/monaco-tailwindcss/pull/14/commits/63d37f7b20a1143292cb8a819abb0b5e52a6d912
https://github.com/remcohaszing/monaco-tailwindcss/pull/14/commits/58105c22a98a2366fd955b0d454768335c9eda05

i found out that this `modifySelectors` is never called (i think unless you have a custom plugin, which we dont allow)

**Comment by remcohaszing** [comment]
No problem! I guess this PR can be closed now. :)

File: types.d.ts [hunk 1]
```diff
@@ -11,15 +11,17 @@ declare module 'tailwindcss/src/lib/generateRules.js' {
 }
 
 declare module 'tailwindcss/src/lib/setupContextUtils.js' {
-  import { TailwindConfig } from 'tailwindcss/tailwind-config';
   import { Container, Root } from 'postcss';
+  import { TailwindConfig } from 'tailwindcss/tailwind-config';
+
+  type ModifierFunction = (options: { className?: string; selector: string }) => string;
```
Comment (mhsdesign) [inline types.d.ts:17]: where the hell did you found that?!!! xD

Comment (remcohaszing) [reply types.d.ts:17]: I based it purely on the implementation in `getVariants`. There’s a good chance I missed several optional properties.

Comment (mhsdesign) [reply types.d.ts:17]: upps ^^ you just put in there the object from the code ^^

**Pull request closed without merging**